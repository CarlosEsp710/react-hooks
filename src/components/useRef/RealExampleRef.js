import React, { useState } from "react";

import { MultipleCustomHooks } from "../examples/MultipleCustomHooks";

import "./ref.css";

export const RealExampleRef = () => {
  const [show, setShow] = useState(false);

  const handleChange = () => setShow(!show);

  return (
    <>
      <h1>RealExampleRef</h1>
      <hr />
      {show && <MultipleCustomHooks />}
      <button className="btn btn-outline-primary mt-5" onClick={handleChange}>
        Show/Hide
      </button>
    </>
  );
};
