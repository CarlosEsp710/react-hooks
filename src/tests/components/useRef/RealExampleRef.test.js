import React from "react";
import { shallow } from "enzyme";

import { RealExampleRef } from "../../../components/useRef/RealExampleRef";

describe("test <RealExampleRef />", () => {
  test("should return component <RealExampleRef />", () => {
    const wrapper = shallow(<RealExampleRef />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(false);
  });

  test("should show component <MultipleCustomHooks />", () => {
    const wrapper = shallow(<RealExampleRef />);
    wrapper.find("button").simulate("click");
    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(true);
  });
});
