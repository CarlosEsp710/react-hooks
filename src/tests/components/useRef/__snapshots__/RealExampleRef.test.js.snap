// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <RealExampleRef /> should return component <RealExampleRef /> 1`] = `
<Fragment>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-outline-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</Fragment>
`;
